contracts/core/RebalancingSetToken.sol

```diff
@@ -312,29 +357,29 @@ contract RebalancingSetToken is
         uint256[] memory outflowUnitArray = new uint256[](combinedTokenArray.length);
 
         // Get bid conversion price
-        uint256 priceNumerator = 1;
-        uint256 priceDivisor = 1;
+        uint256 priceNumerator = 1374;
+
+        // Normalized quantity amount
+        uint256 unitsMultiplier = _quantity.div(minimumBid).mul(auctionPriceDivisor);
 
         for (uint256 i=0; i < combinedTokenArray.length; i++) {
-            uint256 rebalanceUnit = combinedRebalanceUnits[i];
+            uint256 nextUnit = combinedNextSetUnits[i];
             uint256 currentUnit = combinedCurrentUnits[i];
 
             // If rebalance greater than currentUnit*price token inflow, else token outflow
-            if (rebalanceUnit > currentUnit.mul(priceNumerator).div(priceDivisor)) {
-                inflowUnitArray[i] = _quantity.mul(rebalanceUnit.sub(
-                    priceNumerator.mul(currentUnit).div(priceDivisor)
-                )).div(10**18);
+            if (nextUnit > currentUnit.mul(priceNumerator).div(auctionPriceDivisor)) {
+                inflowUnitArray[i] = unitsMultiplier.mul(
```

Comment:
This seems really convoluted. Can we create a method for these calculations?

Reply:
I guess what do you hope to gain by making it a function? It doesn't change the logic, it's pretty much already in its own function, I don't see a need to make another one.

Reply:
An additional function allows us to express the intent of this 5-step arithmetic expression.

IMO, it's really unclear what it is supposed to do. Thus, its harder to audit and ensure the security of it as well.

Reply:
Hmm I'm not sure there's a way to clearly get across exactly what this is supposed to do unless we have a derivation of this formula in the code base. I don't think adding a function will enforce anymore clarity cause they'll still look at this formula and wonder how it was derived. Perhaps an insightful multi-line comment is a better way to go about it?

Reply:
I feel a multi-line doesn't do it justice. you'd have to explain it everywhere you put it.

It seems like the same formula is used twice. With a function, we'd at least enforce that both operate similarly.

The function name itself can be a hint for what the calculations are supposed to be. And we can have a more in-depth function comment in that function too?

Reply:
It's only used here, and I guess I look at the whole thing as one formula. it feels like you're saying let's abstract all five-step functions that have these order of operations. What's important is understanding how token flows are calculated, another function in this case, does not solve that problem I don't beleive.

Reply:
+1, add a private method for this unit array calculation, and you can abstract the formula explanation into that private method. The comments could be improved/some lines are missing some.